runtime/compiler/trj9/optimizer/J9ValuePropagation.cpp

```diff
@@ -406,7 +406,8 @@ J9::ValuePropagation::constrainRecognizedMethod(TR::Node *node)
                                                         TR::VMAccessCriticalSection::tryToAcquireVMAccess);
             if (!getStringlength.hasVMAccess())
                break;
-            len = comp()->fej9()->getStringLength(*stringLocation);
+            uintptrj_t stringObject = comp()->fej9()->getStaticReferenceFieldAtAddress(comp(), (uintptrj_t)stringLocation, receiverChild->getSymbolReference());
```

Comment:
Here `receiverChild->getSymbolReference()` might not be the right symbol reference - e.g. it could be an auto.

If `receiverChildConstraint` is a `VPKnownObject` instead of a `VPConstString`, then there might not even be a static symbol reference that corresponds to the string

Reply:
Hm I see; well since the symref is really only used to get the class, if we don't need to provide a class to the read barrier api, this will become a non-issue.